app/scripts/models/auth_brokers/fx-sync.js

```diff
@@ -77,6 +80,41 @@ define(function (require, exports, module) {
       return proto.initialize.call(this, options);
     },
 
+    fetch (user) {
+      return proto.fetch.call(this, user)
+        .then(() => {
+          if (! this.hasCapability('browserStatus')) {
+            return;
+          }
+
+          // the delay is to give functional tests a little bit of time to attach
+          // WebChannel listeners before continuing.
+          const delay = this.environment.isFunctionalTests() ? 50 : 0;
+
+          // Request the currently signed in user data from the browser.
+          // If the browser returns a `signedInUser`, set that as the
+          // signed in account. If no signed in account is returned,
+          // clear the signed in account.
+          return p().delay(delay)
+            .then(() => this.request(this.getCommand('BROWSER_STATUS')))
+            .then((response = {}) => {
+              const userData = response.signedInUser;
+              if (userData) {
+                // merge account data with any existing account data. A Default account
+                // will be returned if none exists.
+                const account = user.getAccountByUid(userData.uid);
+                account.set(_.pick(userData, 'email', 'sessionToken', 'uid', 'verified'));
+                account.set('sessionTokenContext', SESSION_TOKEN_USED_FOR_SYNC);
+
+                return user.setSignedInAccount(account);
```

Comment:
This persists the data from the handshake into localStorage, right?  I wonder whether that will cause trouble for us when using this in private browsing mode, if the data so written then becomes available to other tabs in the same private-browsing window.  I'll try to build m-c with this patch and find out, but it will take a while as I don't have my dev environment set up on this machine yet...